pkg/config/config_test.go

```diff
@@ -34,6 +39,19 @@ metadata:
 `,
 			expectErrMsg: "`source` must not be empty",
 		},
+		{
+			name: "KCLRun missing matchConstraints",
```

Comment:
Can it be an optional field. When not set the field, it will do nothing on the input resources.